tests/integration/utils.js

```diff
@@ -175,6 +175,26 @@ testUtils.isCouchDB = function (cb) {
   });
 };
 
+testUtils.getServerType = function (cb) {
+  const knownServers = [
+    'couchdb',
+    'express-pouchdb',
+    'pouchdb-express-router',
+  ];
+
+  var {url, options} = parseHostWithCreds(testUtils.couchHost());
+  PouchDB.fetch(url, options).then(function (response) {
+    return response.json();
+  }).then(function (res) {
+    for (const known of knownServers) {
+      if (res[known]) {
+        return cb(known);
+      }
+    }
+    cb('unknown');
+  });
```

Comment:
The last thing I found:

In case of an exception: the callback was never called, resulting in an unreasonable `time out`.
```suggestion
  }).catch(function (_err) {
    cb('unknown');
  });
```

Reply:
Good catch - the error definitely needs handling.  I'd rather throw instead of logging I think.